api/main.go

```diff
@@ -48,21 +47,21 @@ type ApiServer struct {
 
 type MetadataServer struct {
 	address string
-	Logger  *zap.SugaredLogger
+	Logger  logging.Logger
 	meta    pb.MetadataClient
 	client  *metadata.Client
 	pb.UnimplementedApiServer
 	health *health.Health
 }
 
 type OnlineServer struct {
-	Logger  *zap.SugaredLogger
+	Logger  logging.Logger
 	address string
 	client  srv.FeatureClient
 	srv.UnimplementedFeatureServer
 }
 
-func NewApiServer(logger *zap.SugaredLogger, address string, metaAddr string, srvAddr string) (*ApiServer, error) {
+func NewApiServer(logger logging.Logger, address string, metaAddr string, srvAddr string) (*ApiServer, error) {
```

Comment:
super nit: @simba what do you think about making sure the logger is always in the beginning or the end of a function. We should have an order. @ workday the context and logger were always at the end

Reply:
context is always at the beginning in Go, this is the typical design patten (as you can see in a lot of the APIs you use.) We also shouldn't pass the logger in as an argument, but rather a context, in most situations.